operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventFilter.java

```diff
@@ -0,0 +1,58 @@
+package io.javaoperatorsdk.operator.processing.event.internal;
+
+import io.fabric8.kubernetes.client.CustomResource;
+import io.javaoperatorsdk.operator.api.config.ControllerConfiguration;
+
+/**
+ * A functional interface to determine whether resource events should be processed by the SDK. This
+ * allows users to more finely tuned which events trigger a reconciliation than was previously
+ * possible (where the logic was limited to generation-based checking).
+ *
+ * @param <T> the type of custom resources handled by this filter
+ */
+@FunctionalInterface
+public interface CustomResourceEventFilter<T extends CustomResource> {
+
+  /**
+   * Determines whether the change between the old version of the resource and the new one needs to
+   * be propagated to the controller or not.
+   *
+   * @param configuration the target controller's configuration
+   * @param oldResource the old version of the resource
+   * @param newResource the new version of the resource
+   * @return {@code true} if the change needs to be propagated to the controller, {@code false}
+   *         otherwise
+   */
+  boolean acceptChange(ControllerConfiguration<T> configuration, T oldResource, T newResource);
```

Comment:
We have to discuss this @lburgazzoli @metacosm . The old object should not be here IMHO.